docs/src/reference-guide/components/graphql/search-filtering.md

```diff
@@ -0,0 +1,297 @@
+# Search and Filtering
+
+Currently, the Fuel indexer supports the following search and filtering operations:
+
+- ID selection
+- comparison
+- set membership
+- excluding null values
+
+Additionally, you can combine these operations using the `and` or `or` keywords, and invert operations by using the `not` keyword.
+
+To illustrate these operations, we'll use a `filter_example` entity that would typically return the following results with a regular, unfiltered query:
+
+```graphql
+query {
+  filter_example {
+    id
+    foo
+    possibly_null_bar
+    baz
+  }
+}
+```
+
+```json
+{
+  "data": [
+    {
+      "baz": 1,
+      "foo": "beep",
+      "id": 1,
+      "possibly_null_bar": 123
+    },
+    {
+      "baz": 5,
+      "foo": "boop",
+      "id": 2,
+      "possibly_null_bar": null
+    },
+    {
+      "baz": 1000,
+      "foo": "blorp",
+      "id": 3,
+      "possibly_null_bar": 456
+    }
+  ]
+}
+```
+
+> Note: It is important to note that you should practice sensible database design when filtering records. Apply database indicies to the underlying columns in order to make search operations more efficient; however, be advised that an overuse of database indicies will lead to degraded performance.
```

Comment:
Suggested change:
"You should practice sensible database design when filtering records."
This means we avoid repetition of the word "note" here.